src/step_algorithms/dfs.rs

```diff
@@ -0,0 +1,69 @@
+use petgraph::graph::NodeIndex;
+use petgraph::Direction;
+
+use crate::graph::node::NodeState;
+use crate::graph::Graph;
+use crate::step_algorithms::algorithm::{Algorithm, AlgorithmStep, EdgeStep, NodeStep};
+
+impl Algorithm {
```

Comment:
to niedobra architektura - bo nie da się z zewnątrz dodać nowych algorytmów. Musicie to jakoś przedesignować w taki sposób, żeby algorithm był traitem i dało się go zaimplementować, a nie że robicie nową metodę w tym strukcie. Prawdopodobnie będzie wymagało to rozdzielenia "stanu" algorytmu i opisu tego co ma robić, być może jakichś generyków, bo pewnie każdy algorytm będzie pamiętał inne rzeczy. Możemy o tym pogadać dokładniej np. na discordzie gdybyście chcieli